### PR TITLE
feat: added codes to test

### PR DESCRIPTION
**Pull request by tanmoysrt**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -7,6 +7,9 @@ description = "Open Source HR & Payroll Software"
 requires-python = ">=3.10"
 readme = "README.md"
 dynamic = ["version"]
+dependencies = [
+    "frappe-client~=0.1.0.dev"
```
Comment (ruchamahabal) [inline pyproject.toml]: We don't need to add this as a dependency. 
Frappe has frappeclient:
`from frappe.frappeclient import FrappeClient`

https://github.com/frappe/frappe/blob/develop/frappe/frappeclient.py

File: hrms/api/__init__.py [hunk 2]
```diff
@@ -624,3 +625,26 @@ def get_allowed_states_for_workflow(workflow: dict, user_id: str) -> list[str]:
...
 		transition.state for transition in workflow.transitions if transition.allowed in user_roles
 	]
+
+# Notification
+@frappe.whitelist(methods=["GET"])
+def subscribe_push_notification(fcm_token: str):
+	current_user = frappe.session.user
+	data = frappe.db.get_value("User", current_user, ["name"], as_dict=True)
+	username = data.name
+	success, message = FrappeNotification().add_token(username, fcm_token)
```
Comment (ruchamahabal) [inline hrms/api/__init__.py]:
```suggestion
	success, message = FrappeNotification().add_token(frappe.session.user, fcm_token)
```

File: hrms/api/__init__.py [hunk 3]
```diff
@@ -624,3 +625,26 @@ def get_allowed_states_for_workflow(workflow: dict, user_id: str) -> list[str]:
...
+		"success": success,
+		"message": message
+	}
+
+@frappe.whitelist(methods=["GET"])
+def unsubscribe_push_notification(fcm_token: str):
+	current_user = frappe.session.user
+	data = frappe.db.get_value("User", current_user, ["name"], as_dict=True)
+	username = data.name
+	success, message = FrappeNotification().remove_token(username, fcm_token)
```
Comment (ruchamahabal) [inline hrms/api/__init__.py]:
```suggestion
	success, message = FrappeNotification().remove_token(frappe.session.name, fcm_token)
```

frappe.session.user stores the name of the user so can be used directly?

File: hrms/api/__init__.py [hunk 4]
```diff
@@ -624,3 +625,26 @@ def get_allowed_states_for_workflow(workflow: dict, user_id: str) -> list[str]:
 	return [
 		transition.state for transition in workflow.transitions if transition.allowed in user_roles
 	]
+
+# Notification
+@frappe.whitelist(methods=["GET"])
+def subscribe_push_notification(fcm_token: str):
```
Comment (ruchamahabal) [inline hrms/api/__init__.py]: these 2 APIs could be part of FrappeNotification only. Other apps can use them to subscribe/unsubscribe

File: hrms/api/__init__.py (hunk 3 above)
Comment (tanmoysrt) [reply hrms/api/__init__.py]: Yeah we can use that

File: hrms/api/__init__.py (hunk 2 above)
Comment (tanmoysrt) [reply hrms/api/__init__.py]: done !

File: hrms/api/__init__.py (hunk 3 above)
Comment (tanmoysrt) [reply hrms/api/__init__.py]: done !

File: hrms/api/__init__.py (hunk 4 above)
Comment (tanmoysrt) [reply hrms/api/__init__.py]: done !

**Pull request closed without merging**